File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Record.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "records", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "records_unique", columnNames = {"record_name"})
})
public class Record {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "record_id", nullable = false)
    private Long id;

    @NotNull
    @Column(name = "wins", nullable = false)
    private Long wins;

    @NotNull
    @Column(name = "ties", nullable = false)
    private Long ties;

    @NotNull
    @Column(name = "losses", nullable = false)
    private Long losses;

    @NotNull
    @Column(name = "record_name", nullable = false, length = Integer.MAX_VALUE)
    private String recordName;

}

File: umbrella-insurance-react-ui/src/models/records/v1/Record.ts
import { Season } from "../../seasons/v1/Season";

export class Record {
    id?: number;
    wins?: number;
    ties?: number;
    losses?: number;
    recordName?: string;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.wins) {
                this.wins = obj.wins;
            }
            if(obj?.ties) {
                this.ties = obj.ties;
            }
            if(obj?.losses) {
                this.losses = obj.losses;
            }
            if(obj?.recordName) {
                this.recordName = obj.recordName;
            }
        }
    }
}